eng/scripts/OpenAPI Hub/sync-repo-structure-cache.py

```diff
@@ -0,0 +1,118 @@
+from git import Repo
+from pymongo import MongoClient
+from typing import List
+import datetime
+import os
+import re
+
+
+def get_inner_blobs_structure(blob: Repo.tree) -> dict:
+    inner_structure: dict = {}
+    for inner_blob in blob:
+        if inner_blob.type == "tree":
+            inner_structure[inner_blob.name] = get_inner_blobs_structure(inner_blob)
+    return inner_structure
+
+
+def get_repo_tree(repo_url: str, clone_path: str, branch: str) -> [str, dict]:
+    print("Cloning to [{}]".format(clone_path))
+    if not os.path.isdir(clone_path):
+        print("Cloning repo into {} ...".format(clone_path))
+        repo: Repo = Repo.clone_from(repo_url, clone_path, branch=branch)
+    else:
+        print("Repo already cloned, checking out branch: {}".format(branch))
+        repo: Repo = Repo(clone_path)
+        repo.git.checkout(branch, force=True)
+
+    tree: Repo.tree = repo.tree()
+    print("Repo cloned successfully")
+
+    repo_structure: dict = {}
+    for blob in tree:
+        if blob.name == "specification" and blob.type == "tree":
+            repo_structure[blob.name] = get_inner_blobs_structure(blob)
+
+    if "specification" not in repo_structure:
+        raise Exception("No 'specification' folder found in the repo.")
```

Comment:
nit: Probably `RunTimeError` here too. maybe [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError)?

Reply:
Based on the description of the docs, I wouldn't go with `ValueError` since this doesn't happen in an argument. The `repo_structure` is calculated based on the cloned repo. If the cloned repo doesn't have an "specification" directory, it would throw the error. 

I'm going to go with `RunTimeError` here.